tests/test_prettytable.py

```diff
@@ -1196,3 +1196,101 @@ def testBordered(self):
+
+
+class ColumnsMinWidthsTest(unittest.TestCase):
+    def setUp(self):
+        self.x = PrettyTable()
+        self.x.field_names = ["City name", "Area", "Population"]
+        self.x.add_row(["Adelaide", 1295, 1158259])
+        self.x.add_row(["Brisbane", 5905, 1857594])
+        self.x.add_row(["Darwin", 112, 120900])
+
+        self.y = PrettyTable()
+        self.y.add_row(["Adelaide", 1295, 1158259])
+        self.y.add_row(["Brisbane", 5905, 1857594])
+        self.y.add_row(["Darwin", 112, 120900])
+
+        self.auto_index_table = PrettyTable()
+        self.auto_index_table.field_names = ["City name", "Area", "Population"]
+        self.auto_index_table.add_row(["Adelaide", 1295, 1158259])
+        self.auto_index_table.add_row(["Brisbane", 5905, 1857594])
+        self.auto_index_table.add_row(["Darwin", 112, 120900])
+
+        self.add_column_table = PrettyTable()
+        self.add_column_table.field_names = ["City name", "Area", "Population"]
+        self.add_column_table.add_row(["Adelaide", 1295, 1158259])
+        self.add_column_table.add_row(["Brisbane", 5905, 1857594])
+        self.add_column_table.add_row(["Darwin", 112, 120900])
+
+    def testColumnsMinWidths(self):
+        tables = [
+            dict(
+                result="""
++-----------+------+------------+
+| City name | Area | Population |
++-----------+------+------------+
+|  Adelaide | 1295 |  1158259   |
+|  Brisbane | 5905 |  1857594   |
+|   Darwin  | 112  |   120900   |
++-----------+------+------------+
+""",
+                min_widths=[None, None, None],
+            ),
+            dict(
+                result="""
++----------------------+--------------------------------+------------+
+|      City name       |              Area              | Population |
++----------------------+--------------------------------+------------+
+|       Adelaide       |              1295              |  1158259   |
+|       Brisbane       |              5905              |  1857594   |
+|        Darwin        |              112               |   120900   |
++----------------------+--------------------------------+------------+
+""",
+                min_widths=[20, 30, None],
+            ),
+            dict(
+                result="""
++-----------+--------------------------------+------------+
+| City name |              Area              | Population |
++-----------+--------------------------------+------------+
+|  Adelaide |              1295              |  1158259   |
+|  Brisbane |              5905              |  1857594   |
+|   Darwin  |              112               |   120900   |
++-----------+--------------------------------+------------+
+""",
+                min_widths=[2, 30, 3],
+            ),
+        ]
+        for table in tables:
+            self.x.columns_min_widths = table["min_widths"]
+            result = self.x.get_string()
+            self.assertEqual(result.strip(), table["result"].strip())
```

Comment:
Let's stick with pytest style asserts:
```suggestion
            assert result.strip() == table["result"].strip()
```